File: Uebungen/1_Rot13/Java/src/main/java/Rot13Security.java
public class Rot13Security {
    public String Encrypt(String input) {
        if ((input == null) || (input.equals(""))) {
            return "";
        }

        input = input
                .replaceAll("ä", "AE")
                .replaceAll("ö", "OE")
                .replaceAll("ü", "UE")
                .replaceAll("Ä", "AE")
                .replaceAll("Ö", "OE")
                .replaceAll("Ü", "UE")
                .toUpperCase();

        var encrypted = new StringBuilder();
        for (var character : input.toCharArray()) {
            encrypted.append(Shift(character));
        }

        return encrypted.toString();
    }

    private char Shift(char character) {
        var rangeFrom = (int) 'A';
        var rangeTo = (int) 'Z';

        var characterCode = (int) character;

        if (characterCode >= rangeFrom && characterCode <= rangeTo) {
            characterCode += 13;
            if (characterCode > rangeTo) {
                characterCode -= 26;
            }

            character = (char) characterCode;
        }

        return character;
    }
}

File: Uebungen/1_Rot13/C#/Rot13/Rot13Security.cs
using System;
using System.Linq;
using System.Text;

namespace Rot13
{
    public class Rot13Security
    {
        public string Encrypt(string input)
        {
            if (input == null)
                throw new ArgumentNullException();

            if (string.IsNullOrEmpty(input)) 
                return string.Empty;

            input = input
                .ToUpper()
                .Replace("Ä", "AE")
                .Replace("Ö", "OE")
                .Replace("Ü", "UE");

            var encrypted = new StringBuilder();
            foreach (var character in input)
                encrypted.Append(Shift(character));

            return encrypted.ToString();
        }

        private char Shift(char character)
        {
            var range = Enumerable.Range('A', 26).ToArray();
            if (range.Contains(character))
            {
                var shiftedCode = character + 13;
                if (shiftedCode > range.Max()) 
                    shiftedCode -= 26;

                character = (char) shiftedCode;
            }

            return character;
        }
    }
}